advanced_functionality/unity_ml_agents/src/train.py

```diff
@@ -0,0 +1,38 @@
+import argparse
+import numpy as np
+import os
+import tensorflow as tf
+from mlagents_envs.environment import UnityEnvironment
+import mlagents
+import subprocess
+import yaml
+import json
+
+def parse_args():
+    
+    parser = argparse.ArgumentParser()
+
+    # hyperparameters sent by the client are passed as command-line arguments to the script
+    parser.add_argument('--epochs', type=int, default=1)
+    parser.add_argument('--batch_size', type=int, default=64)
+    parser.add_argument('--learning_rate', type=float, default=0.1)
+    
+    # data directories
+    parser.add_argument('--train', type=str, default=os.environ.get('SM_CHANNEL_TRAIN'))
+    parser.add_argument('--test', type=str, default=os.environ.get('SM_CHANNEL_TEST'))
+    parser.add_argument('--env_name', type=str, default=os.environ.get('SM_HP_ENV_NAME'))
+    parser.add_argument('--yaml_file', type=str, default=os.environ.get('SM_HP_YAML_FILE'))
+    #parser.add_argument('--train_config', type=str, default=os.environ.get('SM_HP_TRAIN_CONFIG'))
+    
+    # model directory: we will use the default set by SageMaker, /opt/ml/model
+    parser.add_argument('--model_dir', type=str, default=os.environ.get('SM_MODEL_DIR'))
+    
+    
+    return parser.parse_known_args()
+
+if __name__ == "__main__":
+    args, _ = parse_args()
+    subprocess.call(f'chmod 755 {args.train}/{args.env_name}'.split())    
```

Comment:
If a subprocess call failed, the main process will capture nothing here? e.g. if I passed invalid args, will there be any error trace?

Reply:
Yes, this will catch the error trace and shown in the CW Logs